nifi-commons/nifi-site-to-site-client/src/main/java/org/apache/nifi/remote/client/CachingRemoteGroupContentsManager.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.remote.client;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import org.apache.nifi.remote.util.SiteToSiteRestApiClient;
+import org.apache.nifi.web.api.dto.ControllerDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CachingRemoteGroupContentsManager implements RemoteGroupContentsManager {
```

Comment:
It looks like this code is also in the API client so I'm assuming it should only exist in one place and one of them is probably left-over

Reply:
D'oh! Right you are. Started implementing it in a separate class, then realized that it worked a lot more nicely to just embed the logic in the client and forgot to remove the interface & impl. Will push another commit.